### PR TITLE
feat(server): integrate `server-hstore` into hugegraph

### PR DESCRIPTION
**Pull request by VGalaxies**
subtask of https://github.com/apache/incubator-hugegraph/issues/2265

When introducing hstore, `server-core` needs corresponding modifications. Except for `BytesBuffer.java` in `hugegraph-server/hugegraph-core/src/main/java/org/apache/hugegraph/backend/serializer/`, other files should be consistent with the pd-store branch.

### CODE REVIEW

File: hugegraph-server/hugegraph-core/src/main/java/org/apache/hugegraph/backend/query/Query.java [hunk 1]
```diff
@@ -580,6 +616,13 @@ public static void checkForceCapacity(long count) throws LimitExceedException {
         }
     }
 
+    public enum OrderType {
+        // 批量接口下，返回顺序的要求
```
Comment (javeme) [inline hugegraph-server/hugegraph-core/src/main/java/org/apache/hugegraph/backend/query/Query.java:620]: to be translated

File: hugegraph-server/hugegraph-core/src/main/java/org/apache/hugegraph/backend/serializer/BinaryBackendEntry.java [hunk 2]
```diff
@@ -169,7 +174,10 @@ public boolean mergeable(BackendEntry other) {
             return false;
         }
         if (!this.id().equals(other.id())) {
-            return false;
+            // 兼容从ap查回的数据, vertex id
```
Comment (javeme) [inline hugegraph-server/hugegraph-core/src/main/java/org/apache/hugegraph/backend/serializer/BinaryBackendEntry.java:177]: ditto

File: hugegraph-server/hugegraph-core/src/main/java/org/apache/hugegraph/backend/serializer/BytesBuffer.java [hunk 3]
```diff
@@ -790,6 +790,34 @@ public BinaryId parseId(HugeType type, boolean enablePartition) {
         return new BinaryId(bytes, id);
     }
 
+    /**
+     * 解析 olap id
+     * @param type
+     * @param isOlap
+     * @return
+     */
```
Comment (javeme) [inline hugegraph-server/hugegraph-core/src/main/java/org/apache/hugegraph/backend/serializer/BytesBuffer.java:798]: we can remove unused comments

File: hugegraph-server/hugegraph-core/src/main/java/org/apache/hugegraph/type/HugeTableType.java [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+import java.util.Map;
+
+import org.apache.hugegraph.type.define.SerialEnum;
+
+public enum HugeTableType implements SerialEnum {
+
+    UNKNOWN(0, "UNKNOWN"),
+
+    /* Schema types */
+    VERTEX(1, "V"), // 顶点表
```
Comment (javeme) [inline hugegraph-server/hugegraph-core/src/main/java/org/apache/hugegraph/type/HugeTableType.java:30]: ditto

File: hugegraph-server/hugegraph-hstore/pom.xml [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+    <dependencies>
+        <dependency>
+            <groupId>org.apache.hugegraph</groupId>
+            <artifactId>hugegraph-core</artifactId>
+            <version>${revision}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.hugegraph</groupId>
+            <artifactId>hg-store-client</artifactId>
```
Comment (javeme) [inline hugegraph-server/hugegraph-hstore/pom.xml:40]: prefer hg => hugegraph

File: hugegraph-server/hugegraph-hstore/pom.xml [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+            <version>${revision}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.hugegraph</groupId>
+            <artifactId>hg-pd-client</artifactId>
```
Comment (javeme) [inline hugegraph-server/hugegraph-hstore/pom.xml:45]: prefer hg => hugegraph

File: hugegraph-server/hugegraph-core/pom.xml [hunk 7]
```diff
@@ -253,6 +253,32 @@
...
             <scope>runtime</scope>
         </dependency>
+        <dependency>
+            <groupId>com.google.code.gson</groupId>
+            <artifactId>gson</artifactId>
+            <version>2.8.9</version>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.hugegraph</groupId>
+            <artifactId>hg-pd-client</artifactId>
```
Comment (javeme) [inline hugegraph-server/hugegraph-core/pom.xml:263]: prefer hg => hugegraph

File: hugegraph-server/hugegraph-hstore/pom.xml (hunk 5 above)
Comment (imbajin) [reply hugegraph-server/hugegraph-hstore/pom.xml:40]: > prefer hg => hugegraph

we'll delete all prefix later (include change the name of server)

**Review by imbajin — APPROVED** [review]

**Review by liuxiaocs7 — APPROVED** [review]

**Pull request merged**